pkg/logql/range_vector.go

```diff
@@ -25,16 +25,16 @@ type RangeVectorIterator interface {
 }
 
 type rangeVectorIterator struct {
-	iter                         iter.PeekingSampleIterator
-	selRange, step, end, current int64
-	window                       map[string]*promql.Series
-	metrics                      map[string]labels.Labels
-	at                           []promql.Sample
+	iter                                 iter.PeekingSampleIterator
+	selRange, step, end, current, offset int64
+	window                               map[string]*promql.Series
+	metrics                              map[string]labels.Labels
+	at                                   []promql.Sample
 }
 
 func newRangeVectorIterator(
 	it iter.PeekingSampleIterator,
-	selRange, step, start, end int64) *rangeVectorIterator {
+	selRange, step, start, end, offset int64) *rangeVectorIterator {
```

Comment:
Mirroring previous comment, I don't think we need to make the iterator `offset` aware.